File: app/src/main/java/com/example/narasimhakonapalli_finalassignment/Request/Service.java
package com.example.narasimhakonapalli_finalassignment.Request;

import com.example.narasimhakonapalli_finalassignment.utils.Credentials;
import com.example.narasimhakonapalli_finalassignment.utils.MovieApi;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class Service {
    private static Retrofit.Builder retrofitBuilder =
            new Retrofit.Builder()
                    .baseUrl(Credentials.BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create());

    private static Retrofit retrofit = retrofitBuilder.build();

    private static MovieApi movieApi = retrofit.create(MovieApi.class);

    public MovieApi getMovieApi(){
        return movieApi;
    }

}

File: app/src/main/java/com/example/narasimhakonapalli_finalassignment/Request/Service.kt
package com.example.narasimhakonapalli_finalassignment.Request

import com.example.narasimhakonapalli_finalassignment.utils.Credentials
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import com.example.narasimhakonapalli_finalassignment.utils.MovieApi

object Service {
    private val retrofitBuilder = Retrofit.Builder()
        .baseUrl(Credentials.BASE_URL)
        .addConverterFactory(GsonConverterFactory.create())
    private val retrofit = retrofitBuilder.build()
    val movieApi = retrofit.create(
        MovieApi::class.java
    )
}